packages/lit-node-client-nodejs/src/lib/helpers/session-sigs-validator.ts

```diff
@@ -0,0 +1,210 @@
+  const currentTime = new Date();
+
+  if (isNaN(expirationTime.getTime())) {
+    errors.push(
+      `Invalid Expiration Time format in ${context}: ${expirationTimeStr}`
+    );
+  } else if (expirationTime < currentTime) {
+    errors.push(
+      `Expired ${context}. Expiration Time: ${expirationTime.toISOString()}`
+    );
+  }
+
+  return {
+    isValid: errors.length === 0,
+    errors,
+  };
+}
+
+// Function to parse and validate capabilities
+function parseCapabilities(capabilities: Capability[]): ValidationResult {
+  const errors: string[] = [];
+
+  capabilities.forEach((capability, index) => {
+    const { signedMessage } = capability;
+
+    // Parse the signedMessage
+    const parsedCapabilityMessage = parseSignedMessage(signedMessage);
+    capability.parsedSignedMessage = parsedCapabilityMessage;
+
+    // Extract and validate expiration date
+    const expirationTimeStr = parsedCapabilityMessage['Expiration Time'];
+
+    if (expirationTimeStr) {
+      const validationResult = validateExpiration(
+        expirationTimeStr,
+        `capability ${index}`
+      );
+      if (!validationResult.isValid) {
+        errors.push(...validationResult.errors);
+      }
+    } else {
+      errors.push(
+        `Expiration Time not found in capability ${index}'s signedMessage.`
+      );
+    }
+  });
+
+  return {
+    isValid: errors.length === 0,
+    errors,
+  };
+}
+
+/**
+ * Validates the session signature.
+ *
+ * @param sessionSig - The session signature to validate.
+ * @returns The validation result, indicating whether the session signature is valid and any errors encountered during validation.
+ */
+export function validateSessionSignature(
+  sessionSig: AuthSig
+): ValidationResult {
+  const errors: string[] = [];
+
+  // Parse the main signedMessage
+  let parsedSignedMessage: ParsedSessionMessage;
+  try {
+    parsedSignedMessage = JSON.parse(sessionSig.signedMessage);
+  } catch (error) {
+    errors.push('Main signedMessage is not valid JSON.');
+    return { isValid: false, errors };
+  }
+
+  // Validate capabilities
+  const capabilities: Capability[] = parsedSignedMessage.capabilities;
+
+  if (!capabilities) {
+    errors.push('Capabilities not found in main signedMessage.');
+  } else if (capabilities.length === 0) {
+    errors.push('No capabilities found in main signedMessage.');
+  } else {
+    const capabilitiesValidationResult = parseCapabilities(capabilities);
+
+    if (!capabilitiesValidationResult.isValid) {
+      errors.push(...capabilitiesValidationResult.errors);
+    }
+  }
+
+  // Validate main expiration
+  const mainExpirationTimeStr = parsedSignedMessage['expiration'];
+
+  if (mainExpirationTimeStr) {
+    const validationResult = validateExpiration(
+      mainExpirationTimeStr,
+      'main signedMessage'
+    );
+    if (!validationResult.isValid) {
+      errors.push(...validationResult.errors);
+    }
+  } else {
+    errors.push('Expiration Time not found in main signedMessage.');
```

Comment:
nit: we might want to use wording other than `main` as we refer to these other places with `inner` and `outer`